beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1669,6 +1714,179 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
+            eip4844_fork,
+            at_most_current_epoch.saturating_sub(*MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS),
+        );
+
+        self.try_prune_blobs(force, Some(at_most_data_availability_boundary))
+    }
+
+    /// Try to prune blobs older than the data availability boundary.
+    pub fn try_prune_blobs(
+        &self,
+        force: bool,
+        data_availability_boundary: Option<Epoch>,
+    ) -> Result<(), Error> {
+        let (data_availability_boundary, eip4844_fork) =
+            match (data_availability_boundary, self.spec.eip4844_fork_epoch) {
+                (Some(boundary_epoch), Some(fork_epoch)) => (boundary_epoch, fork_epoch),
+                _ => {
+                    debug!(self.log, "Eip4844 fork is disabled");
+                    return Ok(());
+                }
+            };
+
+        let should_prune_blobs = self.get_config().prune_blobs;
+        if !should_prune_blobs && !force {
+            debug!(self.log, "Blob pruning is disabled";
+                "prune_blobs" => should_prune_blobs
+            );
+            return Ok(());
+        }
+
+        let blob_info = || -> BlobInfo {
+            if let Some(blob_info) = self.get_blob_info() {
+                if blob_info.oldest_blob_slot.epoch(E::slots_per_epoch()) >= eip4844_fork {
+                    return blob_info;
+                }
+            }
+            // If BlobInfo is uninitialized this is probably the first time pruning blobs, or
+            // maybe oldest_blob_info has been initialized with Epoch::default. Start from the
+            // eip4844 fork epoch.
+            BlobInfo {
+                oldest_blob_slot: eip4844_fork.start_slot(E::slots_per_epoch()),
+            }
+        }();
+
+        // todo(emhane): Should we add a marginal for how old blobs we import? If so needs to be
+        // reflected here when choosing which oldest slot to prune from.
+        let oldest_blob_slot = blob_info.oldest_blob_slot;
+        // The last entirely pruned epoch, blobs sidecar pruning may have stopped early in the
+        // middle of an epoch otherwise the oldest blob slot is a start slot.
+        let last_pruned_epoch = oldest_blob_slot.epoch(E::slots_per_epoch()) - 1;
+
+        // At most prune up until the data availability boundary epoch, leaving at least blobs in
+        // the data availability boundary epoch and younger.
+        let end_epoch = {
+            let earliest_prunable_epoch = data_availability_boundary - 1;
+            // Stop pruning before reaching the data availability boundary if a margin is
+            // configured.
+            if let Some(margin) = self.get_config().blob_prune_margin_epochs {
+                earliest_prunable_epoch - margin
+            } else {
+                earliest_prunable_epoch
+            }
+        };
+
+        if !force {
+            if last_pruned_epoch.as_u64() + self.get_config().epochs_per_blob_prune
```

Comment:
There is already a check for the fork further up in that method, but it could mean loads of unnecessary pruning if `epochs_per_blob_prune` isn't constrained to being minimum 1.